packages/front-end/components/Experiment/TabbedPage/HealthTabOnboardingModal.tsx

```diff
@@ -0,0 +1,327 @@
+
+  const source = "health-tab-onboarding";
+
+  if (!dataSource) {
+    throw new Error("Data Source Not Found");
+  }
+  const exposureQuery = dataSource.settings.queries?.exposure?.find(
+    (e) => e.id === experiment.exposureQueryId
+  );
+  if (!exposureQuery) {
+    throw new Error("Exposure Query Not Found");
+  }
+
+  const metadataId = exposureQuery.dimensionSlicesId;
+  const dataSourceId = dataSource.id;
+  const exposureQueryId = exposureQuery.id;
+
+  const [id, setId] = useState<string | null>(
+    exposureQuery.dimensionSlicesId || null
+  );
+  const { data, error, mutate } = useApi<{
+    dimensionSlices: DimensionSlicesInterface;
+  }>(`/dimension-slices/${id}`);
+
+  const setUpHealthTab = async () => {
+    await apiCall(`/organization`, {
+      method: "PUT",
+      body: JSON.stringify({
+        settings: { runHealthTrafficQuery: true },
+      }),
+    });
+    if (
+      id &&
+      data?.dimensionSlices?.results &&
+      data.dimensionSlices.results.length > 0
+    ) {
+      track("Save Dimension Metadata", { source });
+      const updates: Partial<ExposureQuery> = {
+        dimensionSlicesId: id,
+        dimensionMetadata: data.dimensionSlices.results.map((r) => ({
+          dimension: r.dimension,
+          specifiedSlices: r.dimensionSlices.map((dv) => dv.name),
+        })),
+      };
+      await apiCall(`/datasource/${dataSource.id}/${exposureQuery.id}`, {
+        method: "PUT",
+        body: JSON.stringify(updates),
+      });
+    }
+    if (setupChoice === "refresh") {
+      setLoading(true);
+      apiCall<{ snapshot: ExperimentSnapshotInterface }>(
+        `/experiment/${experiment.id}/snapshot`,
+        {
+          method: "POST",
+          body: JSON.stringify({
+            phase,
+          }),
+        }
+      )
+        .then((res) => {
+          trackSnapshot(
+            "create",
+            "HealthTabOnboarding",
+            dataSource?.type || null,
+            res.snapshot
+          );
+
+          setAnalysisSettings(null);
+          // TODO reset analysis settings like baseline row, difference type, etc.
+          mutateSnapshot();
+        })
+        .catch((e) => {
+          console.error(e);
+        });
+    }
+    close();
+    refreshOrganization();
+  };
+
+  useEffect(
+    () =>
+      getLatestDimensionMetadata(
+        dataSourceId,
+        exposureQueryId,
+        metadataId,
+        apiCall,
+        setId,
+        mutate
+      ),
+    [dataSourceId, exposureQueryId, metadataId, setId, apiCall, mutate]
+  );
+
+  if (error) {
+    return <div className="alert alert-error">{error?.message}</div>;
+  }
+  const { status } = getQueryStatus(
+    data?.dimensionSlices?.queries || [],
+    data?.dimensionSlices?.error
+  );
+
+  if (step === 1) {
+    const saveEnabled =
+      id &&
+      status === "succeeded" &&
+      data?.dimensionSlices?.results &&
+      data.dimensionSlices.results.length > 0;
+    return (
+      <>
+        <Modal
```

Comment:
I don't know that each step should be a separate modal. We should ideally be conditionally changing the `children` depending on the step as well as whatever else changes depending on the step (secondary CTA, etc)

Reply:
done ty